cmd/subnetcmd/upgradecmd/apply.go

```diff
@@ -348,3 +361,156 @@ func writeLockFile(precmpUpgrades []params.PrecompileUpgrade, subnetName string)
 
 	return nil
 }
+
+func validateUpgradeBytes(file, lockFile []byte, skipPrompting bool) ([]params.PrecompileUpgrade, error) {
+	upgrades, err := getAllUpgrades(file)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(lockFile) > 0 {
+		lockUpgrades, err := getAllUpgrades(lockFile)
+		if err != nil {
+			return nil, err
+		}
+		match := 0
+		for _, lu := range lockUpgrades {
+			for _, u := range upgrades {
+				if reflect.DeepEqual(u, lu) {
+					match++
+					break
+				}
+			}
+		}
+		if match != len(lockUpgrades) {
+			return nil, errNewUpgradesNotContainsLock
+		}
+	}
+
+	allTimestamps, err := getAllTimestamps(upgrades)
+	if err != nil {
+		return nil, err
+	}
+
+	if !skipPrompting {
+		for _, ts := range allTimestamps {
+			if time.Unix(ts, 0).Before(time.Now()) {
+				ux.Logger.PrintToUser("Warning: a timestamp in the past has been detected. This only works for already deployed upgrades!")
```

Comment:
Reword: `Warning: one or more of your upgrades is set to happen in the past. If you've already upgraded your network, the configuration is likely correct and will not cause problems. If this is a new upgrade, this configuration could cause unpredictable behavior and irrecoverable damage to your Subnet. The config MUST be removed. Use caution before proceeding.` insert appropriate line breaks.